src/Simulation/Simulators/ResourcesEstimator/Simulator.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using Microsoft.Quantum.Simulation.Simulators;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+// using System.IO;
+// using System.Threading.Tasks;
+
+namespace Simulator
+{
+    public class AdvancedSimulator : ResourcesEstimatorWithAdditionalPrimitiveOperations
+    {
+        // public override Task<O> Run<T, I, O>(I args)
+        // {
+        //     var result = base.Run<T, I, O>(args).Result;
+        //     var name = typeof(T).Name;
+        //     File.WriteAllText($"{name}.txt", ToTSV());
+        //     return Task.Run(() => result);
+        // }
+
+        protected override IDictionary<string, IEnumerable<Type>> AdditionalOperations { get; } =
+            new Dictionary<string, IEnumerable<Type>> {
+                ["CCZ"] = new [] { typeof(Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators.Circuits.CCZ) },
+                ["And"] = new [] { typeof(Microsoft.Quantum.Canon.ApplyAnd), typeof(Microsoft.Quantum.Canon.ApplyLowDepthAnd) },
+                ["AdjointAnd"] = Array.Empty<Type>()
+            };
+
+        protected override void InitializeQCTracerCoreListeners(IList<IQCTraceSimulatorListener> listeners)
+        {
+            base.InitializeQCTracerCoreListeners(listeners);
+            tCoreConfig.Listeners.Add(new RuntimeCounter());
+        }
+
+        // CCNOT(a, b, c);
+        // T(a);
+        // T(b);
+
+        // Original QDK ResEst. -> 9 Ts
+        // New QDK ResEst. -> 1 CCZ, 2 Ts
```

Comment:
This comment should contain some context of what it is trying to express. Also referring to _Original_ and _New_ estimator may be misleading.